timed/src/lib.rs

```diff
@@ -95,23 +108,47 @@ impl Trace {
         let start = std::time::Instant::now();
         let mut traces = TRACES.lock().unwrap();
         let entry = traces.entry(id.to_string()).or_insert(vec![]);
+        let mut stats_map = HashMap::new();
+        let mut total_time_nanos: u128 = 0;
 
         processor("[");
         for (i, hop) in entry.iter().enumerate() {
+            if stats == &TracingStats::Statistics {
+                if let Finish(d) = hop.ph {
+                    stats_map.entry(hop.name.clone())
+                        .or_insert(vec![])
+                        .push(d);
+                    total_time_nanos += d.as_nanos();
+                }
+            }
+
             let is_last = i == entry.len() - 1;
             let trace = format!(
-                "{{ \"pid\": 0, \"ts\": {},  \"ph\": \"{:?}\", \"name\": \"{}\" }}",
-                hop.ts, hop.ph, hop.name
+                "{{ \"pid\": 0, \"ts\": {},  \"ph\": \"{}\", \"name\": \"{}\" }}",
+                hop.ts, hop.ph.to_string(), hop.name
             );
             processor(&format!("    {}{}", trace, if !is_last { "," } else { "" }));
         }
         processor("]");
-        processor(&format!("Dumping traces took {:?}", start.elapsed()));
 
         match stats {
             TracingStats::None => {}
-            TracingStats::Percentage => {}
+            TracingStats::Statistics => {
```

Comment:
Can you please sort it descending by total time? I think the most useful out of this is to know which function took longest so you know what you need to optimize.